solr/core/src/java/org/apache/solr/handler/component/SearchHandler.java

```diff
@@ -240,7 +241,7 @@ public void changed(SolrPackageLoader.SolrPackage pkg, Ctx ctx) {
   }
 
   @SuppressWarnings({"unchecked"})
-  private void initComponents() {
+  private void initComponents(boolean isCombinedQuery) {
```

Comment:
This is a code smell - initComponents should not be changing behaviour based on a flag specific to a component.

This would be solved if we inherited from SearchHandler or dynamically injected CombinedQueryComponent using a factory pattern